File: Typescript/Ejercicio 2/AtencionPedidos.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AtencionPedidos = void 0;
var AtencionPedidos = /** @class */ (function () {
    function AtencionPedidos() {
        this.ganancias = 0;
    }
    AtencionPedidos.prototype.atender = function (pedido, gestionPlatos) {
        if (pedido) {
            var precio = gestionPlatos.obtenerPrecio(pedido.plato);
            if (precio !== undefined) {
                this.ganancias += precio;
                console.log("Atendiendo pedido de ".concat(pedido.cliente, " - ").concat(pedido.plato));
            }
        }
        else {
            console.log("No hay pedidos para atender.");
        }
    };
    AtencionPedidos.prototype.mostrarGanancias = function () {
        console.log("Ganancias totales: $".concat(this.ganancias.toFixed(2)));
    };
    return AtencionPedidos;
}());
exports.AtencionPedidos = AtencionPedidos;

File: Go/Ejercicio 2/atencion_pedidos.go
package main

import "fmt"

type AtencionPedidos struct {
	Ganancias float64
}

func (ap *AtencionPedidos) Atender(pedido *Pedido, gestionPlatos *GestionPlatos) {
	if pedido != nil {
		precio := gestionPlatos.ObtenerPrecio(pedido.Plato)
		ap.Ganancias += precio
		fmt.Printf("Atendiendo pedido de %s - %s\n", pedido.Cliente, pedido.Plato)
	} else {
		fmt.Println("No hay pedidos para atender.")
	}
}

func (ap *AtencionPedidos) MostrarGanancias() {
	fmt.Printf("Ganancias totales: $%.2f\n", ap.Ganancias)
}
